tinker/faculty_bios/faculty_bio_controller.py

```diff
@@ -122,21 +124,201 @@ def _iterate_child_xml(self, child, author):
+                if soup.find('publications').text:
+                    publications = self.strip_html_tags(unicode(soup.find('publications').renderContents('utf-8', True), "utf-8"))
+                presentations = ""
+                if soup.find('presentations').text:
+                    presentations = self.strip_html_tags(unicode(soup.find('publications').renderContents('utf-8', True), "utf-8"))
+                certificates = ""
+                if soup.find('certificates').text:
+                    certificates = self.strip_html_tags(unicode(soup.find('certificates').renderContents('utf-8', True), "utf-8"))
+                organizations = ""
+                if soup.find('organizations').text:
+                    organizations = self.strip_html_tags(unicode(soup.find('organizations').renderContents('utf-8', True), "utf-8"))
+                hobbies = ""
+                if soup.find('hobbies').text:
+                    hobbies = self.strip_html_tags(unicode(soup.find('hobbies').renderContents('utf-8', True), "utf-8"))
+
+                # This is the initial dictionary of what will be returned
+                page_values = {
+
+                    'first': child.find('.//first').text or "",
+                    'last': child.find('.//last').text or "",
+                    'author': author_text,
+                    'location': soup.find('faculty_location').text,
+                    'highlight': soup.find('highlight').text,
+                    'job-titles': soup.find('job-titles').text,
+                    'email': soup.find('email').text,
+                    'started-at-bethel': soup.find('started-at-bethel').text,
+                    'education': soup.find('education').text,
+                    'biography': biography,
+                    'courses': courses,
+                    'awards': awards,
+                    'publications': publications,
+                    'presentations': presentations,
+                    'certificates': certificates,
+                    'organizations': organizations,
+                    'hobbies': hobbies,
+                    'areas': soup.find('areas').text,
+                    'research-interests': soup.find('research-interests').text,
+                    'teaching-specialty': soup.find('teaching-specialty').text,
+                    'quote': soup.find('quote').text,
+                    'website': soup.find('website').text
+                }
+                # Add all the job-titles and education dictionaries to the main dictionary being returned
+                for dicts in my_list:
+                    page_values.update(dicts)
+                # Add the max_dict which holds both the maximum number of jobs and the maximum number of educations to
+                # the main dictionary
+                page_values.update(max_dict)
+
+            # Returns the page_values
             return page_values
         else:
             return None
 
+    def strip_html_tags(self, html):
```

Comment:
I don't know about this method. I feel like we can handle this in a better way - as this is super complex. If you are trying to strip html tags, element tree has a tostring() method.

Reply:
Removed method. In process of formatting it better.

Reply:
Done